src/enc/lzma_writer.rs

```diff
@@ -85,6 +85,11 @@ impl<W: Write> LzmaWriter<W> {
         Self::new(out, options, false, use_end_marker, None)
     }
 
+    /// Returns a wrapper around `self` that will finish the stream on drop.
+    pub fn auto_finish(self) -> AutoFinisher<Self> {
```

Comment:
I think we should also implement the auto_finish() function for the LZMA2 writer, so that the API is the same for all single threaded writer.